packages/sdk-core/src/common/DelayHelper.ts

```diff
@@ -6,12 +8,21 @@ export class Delay {
      */
     public static wait(timeout: number, signal?: AbortSignal) {
         return new Promise<void>((resolve, reject) => {
+            // intervalId has to be defined here,
+            // as abortCallback can execute before setTimeout is executed
+            // eslint-disable-next-line prefer-const
+            let intervalId: NodeJS.Timeout | undefined;
```

Comment:
does it work correctly with types in the browser?  I think you can use `ReturnType<typeof setTimeout>` instead

Reply:
as long as it builds, it's fine. And it builds 😄

Reply:
How about my proposal? I believes it fits better to our code. Wdyt?

Reply:
It makes no difference at all, but changed.